malaria24/ona/tests/test_models.py

```diff
@@ -400,16 +402,57 @@ def tearDown(self):
             new_case_alert_case_investigators, sender=ReportedCase)
 
     def get_week(self, cases):
-        start_date = cases \
+        test_cases = ReportedCase.objects.all().order_by("create_date_time")
+        start_date = test_cases \
             .first().create_date_time.strftime(
                 "%d %B %Y"
             )
-        end_date = cases \
+        end_date = test_cases \
             .last().create_date_time.strftime(
                 "%d %B %Y"
             )
         return "{0} to {1}".format(start_date, end_date)
 
+    @responses.activate
+    def test_overall_week_ranges(self):
+        self.mk_actor(role=MANAGER_NATIONAL,
+                      email_address='manager@example.org')
+        ehp1 = self.mk_ehp(name='EHP1', email_address='ehp1@example.org')
+        ehp2 = self.mk_ehp(name='EHP2', email_address='ehp2@example.org')
+        self.mk_mis(name='MIS', email_address='mis@example.org')
+
+        for i in range(10):
+            case = self.mk_case(gender='female', facility_code='342315')
+            case.create_date_time = (datetime(2018, 2,
+                                              random.choice(range(1, 5)),
```

Comment:
I think since we're using `random` here, there's a chance that we won't get `1` as the day in any of the generated `datetime`'s. I think we'll need to set these datetimes to something not random to ensure that the test passes every time. eg. for the first look, set the day to `i + 1`, and in the second loop, set it to `i + 5`, that way our date range is guaranteed to be from `1` to `14`.